### ISSUE TITLE
Rope/Water Online GameWindow Crash Bug

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmith/hedgewars#366]
```
What steps will reproduce the problem?
1. Open a room and begin game online (by yourself or with others)
2. Activate rope utility right as/before you enter any water zone at bottom of 
map
3. easiest to see on octorama, but is problem on any map.

What is the expected output? What do you see instead?

When rope tool deployed right before entering water zones on online server 
games, the game window for rope deploying user gets closed and user is left in 
room lobby(not main lobby), with other users in-game experiencing lag until 
crashed user is kicked. 

What version of the product are you using? On what operating system?

0.9.17 Mac osx 10.5.8

Please provide any additional information below.
demo and log attached

```

-----
Original issue reported on code.google.com by `Logan.sh...@gmail.com` on 29 Feb 2012 at 2:05

Attachments:
* [ropewatercrash.zip](https://storage.googleapis.com/google-code-attachments/hedgewars/issue-366/comment-0/ropewatercrash.zip)

**Comment by GoogleCodeExporter** [comment]
```
Heh. I kinda missed the "Octorama" part until I played the demo back.

This should really *only* happen on Underwater or Compost themes.

Is due to the fact that there's no need for drowning animation on those, so the 
hedgehog gear is removed immediately to keep the game moving.  Exposes some 
carelessness in the rope code.

They are actually the fastest themes to play :)

Anyway. Should be easy to fix, and even backwards compatible, although is 
unlikely we'll push a patch release for this, so unless you do your own 
builds...

Workaround is avoiding those themes I suppose.

```

-----
Original comment by `kyberneticist@gmail.com` on 29 Feb 2012 at 11:19

**Comment by GoogleCodeExporter** [comment]
```
Extra if test tossed in.  Fixed in rev 744b9430557a

```

-----
Original comment by `kyberneticist@gmail.com` on 29 Feb 2012 at 11:25
* Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
-----
Original comment by `bmarti...@gmail.com` on 27 Nov 2012 at 6:47
* Changed state: **Verified**
